use_case_guide/real_time_dashboards.rst

```diff
@@ -282,74 +237,87 @@ which often appear.
+A common question in HTTP analytics deals with :ref:`approximate distinct counts
 <count_distinct>`: How many unique visitors visited your site over the last month?
-Answering this question exactly requires storing the list of all previously-seen visitors
-in the rollup tables, a prohibitively large amount of data. A datatype called hyperloglog,
-or HLL, can answer the query approximately; it takes a surprisingly small amount of space
-to tell you approximately how many unique elements are in a set you pass it. Its accuracy
-can be adjusted. We'll use ones which, using only 1280 bytes, will be able to count up to
-tens of billions of unique visitors with at most 2.2% error.
+Answering this question *exactly* requires storing the list of all previously-seen visitors
+in the rollup tables, a prohibitively large amount of data. However an approximate answer
+is much more manageable.
+
+A datatype called hyperloglog, or HLL, can answer the query
+approximately; it takes a surprisingly small amount of space to tell you
+approximately how many unique elements are in a set. Its accuracy can be
+adjusted. We'll use ones which, using only 1280 bytes, will be able to
+count up to tens of billions of unique visitors with at most 2.2% error.
 
 An equivalent problem appears if you want to run a global query, such as the number of
-unique ip addresses which visited any of your client's sites over the last month. Without
-HLLs this query involves shipping lists of ip addresses from the workers to the coordinator for
+unique IP addresses which visited any of your client's sites over the last month. Without
+HLLs this query involves shipping lists of IP addresses from the workers to the coordinator for
 it to deduplicate. That's both a lot of network traffic and a lot of computation. By using
 HLLs you can greatly improve query speed.
 
-First you must install the hll extension; `the github repo
+First you must install the HLL extension; `the github repo
 <https://github.com/aggregateknowledge/postgresql-hll>`_ has instructions. Next, you have
 to enable it:
 
 .. code-block:: sql
 
-  -- this part must be run on all nodes
-  CREATE EXTENSION hll;
-
-  -- this part runs on the coordinator
-  ALTER TABLE http_request_1min ADD COLUMN distinct_ip_addresses hll;
+  -- Run on all nodes ------------------------------------
 
-To make hll functions work with distributed tables, we should define hll aggregate functions
-with the name `sum` on all nodes and let Citus use these functions for distributed queries:
+  CREATE EXTENSION hll;
 
-.. code-block:: sql
+  -- Run on the coordinator node -------------------------
 
-  -- alias for hll_add_agg
+  -- allow SUM to work on hashvals (alias for hll_add_agg)
   CREATE AGGREGATE sum(hll_hashval) (
-       SFUNC = hll_add_trans0,
-       STYPE = internal,
-       FINALFUNC = hll_pack
+    SFUNC = hll_add_trans0,
+    STYPE = internal,
+    FINALFUNC = hll_pack
   );
 
-  -- alias for hll_union_agg
```

Comment:
Did we not need this here for the first example?

Reply:
The modification to `http_request_1min` in this example uses `SUM(hll_hash_text(ip_address))`. But I've been testing only on Cloud which has done all this stuff for me, so I'll check this guide on a local cluster and verify exactly when we need to create custom aggregates.

Reply:
Verified locally that we do need the custom aggregate.

Also discovered it must be created on all nodes, not just the coordinator, and updated the comment in the code block.

Reply:
To test, just now I created a new dev formation. It has hll 2.10 installed. Then:

```
❯ \da
                                                             List of aggregate functions
┌────────┬────────────────┬──────────────────────────────┬────────────────────────────────────────────────┬─────────────────────────────────────────┐
│ Schema │      Name      │       Result data type       │              Argument data types               │               Description               │
├────────┼────────────────┼──────────────────────────────┼────────────────────────────────────────────────┼─────────────────────────────────────────┤
│ public │ hll_add_agg    │ hll                          │ hll_hashval                                    │                                         │
│ public │ hll_add_agg    │ hll                          │ hll_hashval, integer                           │                                         │
│ public │ hll_add_agg    │ hll                          │ hll_hashval, integer, integer                  │                                         │
│ public │ hll_add_agg    │ hll                          │ hll_hashval, integer, integer, bigint          │                                         │
│ public │ hll_add_agg    │ hll                          │ hll_hashval, integer, integer, bigint, integer │                                         │
│ public │ hll_union_agg  │ hll                          │ hll                                            │                                         │
│ public │ max            │ citext                       │ citext                                         │                                         │
│ public │ min            │ citext                       │ citext                                         │                                         │
│ public │ sum            │ group_elements_size_result[] │ group_elements_size_result[]                   │                                         │
│ public │ sum            │ hll                          │ hll                                            │                                         │
│ public │ sum            │ hll                          │ hll_hashval                                    │                                         │
│ public │ sum            │ integer[]                    │ integer[]                                      │                                         │
│ public │ topn_add_agg   │ jsonb                        │ text                                           │ aggregate the items into one counter    │
│ public │ topn_union_agg │ jsonb                        │ jsonb                                          │ aggregate the counters into one counter │
└────────┴────────────────┴──────────────────────────────┴────────────────────────────────────────────────┴─────────────────────────────────────────┘
```